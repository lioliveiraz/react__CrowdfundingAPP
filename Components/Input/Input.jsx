import React from 'react';

function Input(props) {
    const { value, handleChange, type, name, placeholder, label, max } = props;


    return (
        <>
            <label htmlFor={name}>{label}</label>
            <input
                data-testid="input"
                value={value}
                onChange={(e) => handleChange(e.target.value, e.target.name)}
                name={name}
                type={type}
                placeholder={placeholder}
                max={max}
                min={10}
                autoFocus
            />
        </>
    );
}
export default Input;