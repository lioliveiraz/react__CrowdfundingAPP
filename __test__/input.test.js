import Input from './../Components/Input/Input';
import { shallow } from 'enzyme';


describe('input', () => {
    let wrapper, userInput;
    userInput = {
        value: "", handleChange: "", type: "", name: "", placeholder: "", label: "", max: ""
    };
    beforeEach(() => {
        wrapper = shallow(<Input userInput={userInput}></Input>);
    });
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
        /*   expect(wrapper.getElement(<label />)).toEqual(); */
    });

});