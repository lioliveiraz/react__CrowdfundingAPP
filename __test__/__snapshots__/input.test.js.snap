// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input check render should render without props correctly 1`] = `ShallowWrapper {}`;
